### PR TITLE
Problem Set 3 (in progress)

### PR DESCRIPTION
**Pull request by stonedas**
This is what I have for Problem set 3 so far, I haven't done question 5 yet because I wanted to get feedback on 1-4 before I attempted plotting the data.

### CODE REVIEW

**Comment by lkwhite** [comment]
Hey, sorry for not getting to this sooner. Could you either commit your knitted html file or let us know what specific issues you're having? It's harder to review your code without being able to see the output.

**Comment by stonedas** [comment]
I could not figure out how to commit the HTML but I did publish it to RPubs and I included the URL in my most recent commit. But I'll put it here as well: http://rpubs.com/stonedas/256869 

I am not having any specific issues, I was just hoping that you could look over what I have done and give me feedback before I attempt number 5. Thanks!

**Comment by lkwhite** [comment]
Thanks, I will take a look!

File: README.Rmd [hunk 1]
```diff
@@ -100,7 +104,10 @@ raw_data | awk '$4 == "6hpi"' | sort -k6nr | head -n 1 | cut -f2
 ```
 
 ```{r}
-# translate to dplyr code
+library(stringr)
+cleaned_data %>% mutate(time.point = str_replace(time.point, 'hpi', ''), time.point = str_c('hpi.', time.point)) %>% spread(time.point, value) -> cleaned_data_hpi
+
+arrange(cleaned_data_hpi, hpi.6) %>% tail() %>% slice(1) %>% select(plant, fungus)
```
Comment (mandylr) [inline README.Rmd]: This makes sense with how your data is structured, but can you do this with tidy data? Also, you just want the plant column, not plant and fungus columns.

File: README.Rmd [hunk 2]
```diff
@@ -100,7 +104,10 @@ raw_data | awk '$4 == "6hpi"' | sort -k6nr | head -n 1 | cut -f2
 ```
 
 ```{r}
-# translate to dplyr code
+library(stringr)
+cleaned_data %>% mutate(time.point = str_replace(time.point, 'hpi', ''), time.point = str_c('hpi.', time.point)) %>% spread(time.point, value) -> cleaned_data_hpi
```
Comment (mandylr) [inline README.Rmd]: This is useful for what you're doing, but this is not clean data after you ```spread``` the time.point and value columns. How could you make this tidy again?

File: README.Rmd [hunk 3]
```diff
@@ -110,7 +117,7 @@ raw_data | awk '$4 == "18hpi"' | sort -k6nr | head -n 1 | cut -f2,3
 ```
 
 ```{r}
-# translate to dplyr code
+arrange(cleaned_data_hpi, hpi.18) %>% tail() %>% slice(1) %>% select(plant, fungus)
```
Comment (mandylr) [inline README.Rmd]: again, try this with clean data.

File: README.Rmd [hunk 4]
```diff
@@ -138,12 +143,21 @@ cleaned_data %>%
 4. Sort by these statistics and use the `dplyr` verb `slice` to pull the ones you want (i.e., the top 3). Note you will have to remove `gene.name` from the grouping so that sorting works.
 
 ```{r problem_3}
+cleaned_data_hpi %>% mutate(expr.diff = hpi.24 - hpi.6) -> expr_data
+
+expr_data %>% group_by(gene.name, plant, fungus) %>% summarise(mean_expr = mean(expr.diff), var_exp = var(expr.diff)) %>% ungroup(gene.name) %>% arrange(mean_expr, var_exp) %>% group_by(plant, fungus) %>% tail %>% slice(1:3)
```
Comment (mandylr) [inline README.Rmd]: When you arrange, you want the highest expression difference and the lowest variance, so you need to include ```desc()``` for one of the two. Also, there's not need to use tail and slice. You should have a table with 12 rows (3 for each plant/fungus group). I think your tail code is messing this up, giving you only 6 rows.

File: README.Rmd [hunk 5]
```diff
@@ -138,12 +143,21 @@ cleaned_data %>%
...
+
+expr_data %>% group_by(gene.name, plant, fungus) %>% summarise(mean_expr = mean(expr.diff), var_exp = var(expr.diff)) %>% ungroup(gene.name) %>% arrange(mean_expr, var_exp) %>% group_by(plant, fungus) %>% tail %>% slice(1:3) -> con_diff_data
+
 ```
 
 ## Problem 4
 
 Now examine the above final data frame above and write a few sentences putting inline code in least 3 places. For example, There are `r nrow(mtcars)` rows of data in `mtcars`.
 
+There are `r nrow(con_diff_data)` rows and `r ncol(con_diff_data)` columns in `con_diff_data`. There are `r n_groups(con_diff_data)` combinations of plant and fungus in `con_diff_data`.
```
Comment (mandylr) [inline README.Rmd]: There should be 4 groups of plants/fungus but your code here looks right, I think there's a problem with your ```con_diff_data``` tibble.

**Comment by stonedas** [comment]
Thanks for the suggestions, I am currently trying to plot the data but I can't figure out how to filter out the 3 genes. I can filter one of them but I don't know how to say "or" for the filter function.

**Comment by mandylr** [comment]
An easier way to do it is use one of the join functions. Check out what ```semi_join()``` can do for you.

**Comment by mandylr** [comment]
Also, [here](http://stackoverflow.com/questions/25647470/filter-multiple-conditions-dplyr) is an answer for how to use "or" with filter, but it's more complicated than using a join function in this problem.

**Comment by stonedas** [comment]
I added another commit, I'm still working on problem 5 though.

**Comment by stonedas** [comment]
I believe I also updated the HTML on R pubs 
http://rpubs.com/stonedas/256869

**Comment by mandylr** [comment]
Nice work. Finish up the plot and you're good to go!

**Comment by stonedas** [comment]
I added another commit with what I have for question 5, I've done everything with that question that I have been able to figure out. I can't figure out how to facet it by "plant" and "bgh" and I don't know how to do the replicate part. Thanks for the help, sorry I'm so stuck on Problem Set 3, I'm trying to focus more on Problem Set 4 now.

**Comment by mandylr** [comment]
[facet quick guide](http://sape.inf.usi.ch/quick-reference/ggplot2/facet): check out the ```facet_grid()``` usage.

For the replicate part: [point shapes](http://www.sthda.com/english/wiki/ggplot2-point-shapes). Can you figure out how they changed the shape based on cycle in the scatter plots with multiple groups section?

**Comment by stonedas** [comment]
So I still can't seem to get this figured out. I may just plan on coming into office hours sometime and get help with this. I know its technically spring break this week and we don't have class but would you still be available some time this week to meet and work on this with me? I really appreciate all your help so far.

**Comment by lkwhite** [comment]
I think you're really close. We can do drop-ins as long as you're ok waiting if both Mandy & I are in the middle of something. 10th floor of RC1S, first lab space after you come in from the elevators.

**Comment by stonedas** [comment]
I think I have everything finished with problem set 3 now, I added another commit yesterday.

**Comment by jayhesselberth** [comment]
👍  strong work

**Pull request closed without merging**